.github/workflows/release.yml

```diff
@@ -1,16 +1,17 @@
 name: Build distribution
 
 on:
-  push:
+  workflow_run:
+    workflows: [ "CI" ]
     branches: [ main ]
-  pull_request:
+    types:
+      - completed
   release:
     types: [ published ]
 
 jobs:
-  test:
+  publish:
```

Comment:
How about you make a duplicate job in this file?

Call this one `publish-test`, the subsequent one `publish` then `needs: publish-test`.

([Docs link](https://docs.github.com/en/actions/using-jobs/using-jobs-in-a-workflow#example-requiring-successful-dependent-jobs))

`publish-test` publishes to Test PyPI, the latter to the actual PyPI repo.